cryptographic_estimators/MQEstimator/MQAlgorithms/booleansolve_fxl.py

```diff
@@ -32,38 +32,38 @@
 
 
 class BooleanSolveFXL(MQAlgorithm):
-    """
-    Construct an instance of BooleanSolve and FXL estimator
-
-    BooleanSolve and FXL are algorithms to solve the MQ problem over GF(2) and GF(q), respectively [BFSS11]_ [CKPS]_.
-    They work by guessing the value of $k$ variables and computing the consistency of the resulting subsystem.
-
-    INPUT:
-
-    - ``problem`` -- MQProblem object including all necessary parameters
-    - ``w`` -- linear algebra constant (2 <= w <= 3) (default: 2.81)
-    - ``h`` -- external hybridization parameter (default: 0)
-    - ``memory_access`` -- specifies the memory access cost model (default: 0, choices: 0 - constant, 1 - logarithmic, 2 - square-root, 3 - cube-root or deploy custom function which takes as input the logarithm of the total memory usage)
-    - ``complexity_type`` -- complexity type to consider (0: estimate, 1: tilde O complexity, default: 0)
-
-    NOTE:
-        - For the memory complexity of `las_vegas` variant, which is XL with the block Wiedemann algorithm, this module
-        follows the analysis in [BBC+22]_ (Section 8). There it is stated that any row of the Macaulay matrix can be
-        build on the fly. Hence the memory demand of this algorithm is dominated by the memory needed to store two
-        vectors of length `N` over `GF(q)`, where N is the number of columns of the Macaulay Matrix.
-
-    EXAMPLES::
-
-        sage: from cryptographic_estimators.MQEstimator.MQAlgorithms.booleansolve_fxl import BooleanSolveFXL
-        sage: from cryptographic_estimators.MQEstimator.mq_problem import MQProblem
-        sage: E = BooleanSolveFXL(MQProblem(n=10, m=12, q=7))
-        sage: E
-        BooleanSolveFXL estimator for the MQ problem with 10 variables and 12 polynomials
-    """
-
     _variants = (MQ_LAS_VEGAS, MQ_DETERMINISTIC)
-
     def __init__(self, problem: MQProblem, **kwargs):
+        """Construct an instance of BooleanSolve and FXL estimator.
+
+        BooleanSolve and FXL are algorithms to solve the MQ problem over GF(2) and GF(q), respectively [BFSS11]_ [CKPS]_.
+        They work by guessing the value of k variables and computing the consistency of the resulting subsystem.
+
+        Args:
+            problem (MQProblem): Object including all necessary parameters.
+            w (float): Linear algebra constant (2 <= w <= 3). Default is 2.81.
+            h (int): External hybridization parameter. Default is 0.
+            memory_access (int): Specifies the memory access cost model.
+                0 - constant (default)
+                1 - logarithmic
+                2 - square-root
+                3 - cube-root
+                Alternatively, deploy a custom function which takes as input the logarithm of the total memory usage.
```

Comment:
and returns the logarithm of the memory access cost.

Reply:
Done.